packages/medusa/src/api/routes/admin/orders/cancel-fulfillment.ts

```diff
@@ -75,9 +83,18 @@ export default async (req, res) => {
 
   const manager: EntityManager = req.scope.resolve("manager")
   await manager.transaction(async (transactionManager) => {
-    return await orderService
+    await orderService
       .withTransaction(transactionManager)
       .cancelFulfillment(fulfillment_id)
+
+    const fulfillment = await fulfillmentService
+      .withTransaction(transactionManager)
+      .retrieve(fulfillment_id, { relations: ["items", "items.item"] })
+
+    await adjustInventoryForCancelledFulfillment(
+      fulfillment,
+      productVariantInventoryService.withTransaction(transactionManager)
```

Comment:
suggestion, thought: I think a context param would be more elegant to carry services alongside relevant options + would be a more scalable API

Reply:
Very non-blocking comment. Just passing by

Reply:
Great idea! It will make a lot of sense when we also refactor transaction-steps into a separate folder Added the context here and for "create-fulfillment" as well 👍